0017-chart-namespace-overrides/README.md

```diff
@@ -0,0 +1,233 @@
+the ZEP is going to cover.
+-->
+
+<!--
+A table of contents helps reviewers quickly navigate the ZEP and highlights
+any additional information provided beyond the standard ZEP template.
+-->
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [User Stories (Optional)](#user-stories-optional)
+    - [Story 1](#story-1)
+    - [Story 2](#story-2)
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Design Details](#design-details)
+  - [Test Plan](#test-plan)
+      - [Prerequisite testing updates](#prerequisite-testing-updates)
+      - [Unit tests](#unit-tests)
+      - [e2e tests](#e2e-tests)
+  - [Graduation Criteria](#graduation-criteria)
+  - [Upgrade / Downgrade Strategy](#upgrade--downgrade-strategy)
+  - [Version Skew Strategy](#version-skew-strategy)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Infrastructure Needed (Optional)](#infrastructure-needed-optional)
+<!-- /toc -->
+
+## Summary
+
+This ZEP proposes to enable a namespace override for charts similar to the namespace override [functionality available in UDS CLI](https://uds.defenseunicorns.com/reference/bundles/overrides/#namespace).  This would allow namespaces of charts within a Zarf package to be overridden so that multiples of the same Zarf package could be deployed to the same cluster under different namespaces (without needing to maintain variants of the same package).
+
+## Motivation
+
+Doing this allows more flexibility with certain Zarf packages where you may want to have multiples of them installed in the cluster with slightly different configurations (such as [GitLab Runners](https://github.com/defenseunicorns/uds-package-gitlab-runner)).  Right now the release namespace of any chart has to be hardcoded into the package and will be overwritten even if the chart allows namespace overrides for some objects within the chart.  The current behavior is also different from what Helm does by default which may not be what users of Zarf expect (Helm allows the use of the `namespace` flag on install to set the Chart's namespace without it needing to be baked into the Chart).
+
+### Goals
+
+- Provide a way for a Zarf package containing Helm Charts to be easily installed more than once with different configurations
+
+### Non-Goals
+
+- Move away from the declarative nature of Zarf packages
+
+## Proposal
+
+The proposed solution is to add a deploy configuration option that has a `map[string]string` referencing the names of charts within the Zarf Package correlated with their overridden namespaces.
+
+### User Stories (Optional)
+
+#### Story 1
+
+**As** Jacquline **I want** to be able to set namespace overrides **so that** I can install the same package with different configurations in different namespaces.
+
+##### Option 1 (Flatter Namespaces)
```

Comment:
I like option 1 the best,  I feel namespaces should be a first class citizen

Reply:
Why do we want to treat namespace specially from any other variable?

Reply:
Zarf variables are templated into manifests, values-files, and helm-charts. For example, if you had the variable `###ZARF_VAR_MY_IMAGE_TAG###` in your values file and the value was set to v0.0.1, when Zarf deploys that manifest it will change the value to v0.0.1. My worry is if we also make variables a way to define namespaces based on the name of the variable, it will feel magical.

Reply:
Now that this is more based on https://github.com/zarf-dev/proposals/pull/22/files I added some context on why there needs to be some other canonical value - basically the Zarf package secret in the cluster needs to be unique per package name and namespace - if it is not namespaced it will be overwritten and Zarf will forget about the old package.